src/stores/game.js

```diff
@@ -782,5 +782,13 @@ export const useGameStore = defineStore('game', {
       }
     },
 
+    isUserAPlayer() {
+      return this.userId === this.p0?.id || this.userId === this.p1?.id;
+    },
+
+    getMyOriginalPlayerNumber() {
+      return this.userId === this.p0?.id ? 0 : 1;
+    },
```

Comment:
`userId` isn't defined on the game store. In fact we don't currently store the user's id on the client at all, though perhaps we should.

The `game` store has `resetPNumIfNull()` which is how we currently set the pNum when it's missing. That function currently sets your pNum to the matching player's index by username if you're a player in the game, and it sets it to 0 if you're spectating. We should update that function to account for the pNum query parameter if it is set if you're also spectating. Then I think we could rely on that function instead these new ones.